backend/main/types.mo

```diff
@@ -7,6 +7,7 @@ module {
 
 	public type Dependency = {
 		name: PackageName; // max 50
+		repo: Text; // max 250
```

Comment:
This will most likely break the canister and put it at risk of losing its data during an upgrade. 
I think the best approach would be to back up the data beforehand so it can be replaced after the upgrade.

[reference](https://forum.dfinity.org/t/where-is-the-danger-zone-for-projects-written-in-motoko-and-deployed-on-the-ic/11654/3?u=tomijaga)

Reply:
That's cool, I didn't know you could use those system functions for updating a collection's type. 
I'll try it out and share the results.

Reply:
I spent a little time looking for a solution that required little changes to the rest of the code. 
The solution I came up with was to change the names of the previous collections and types so that the new collections and types could retain the old names after the migration. With this method, we can keep the types and collections names we are used to and not have to update the code too much.

I added `V1` to the end of the [previous types](https://github.com/ZenVoich/mops/pull/6/commits/e90dffecf23225ae3d01ac89213fb4dd2ffd286a#diff-fc554db85240661a102dda020968bc87ab7cd75f1a59954b000503fc10397d6aR8-L52) and added `prev` to the start of the [previous collections](https://github.com/ZenVoich/mops/pull/6/files#diff-5d5eb7c0626ae09b21ba7d6bd3e362aa292613a1aa379fc032c98d296b23a658R46-R50)

Here's the [migration function](https://github.com/ZenVoich/mops/pull/6/files#diff-5d5eb7c0626ae09b21ba7d6bd3e362aa292613a1aa379fc032c98d296b23a658R482-R522) converting the data in the preupgrade function.

Reply:
> Here's the [migration function](https://github.com/ZenVoich/mops/pull/6/files#diff-5d5eb7c0626ae09b21ba7d6bd3e362aa292613a1aa379fc032c98d296b23a658R482-R522) converting the data in the preupgrade function.

I'm afraid it wouldn't work. If you change stable vars name or type structure - data will be lost.

The `preupgrade` method is called for the previous canister code, then canister is upgraded(with preserving stable vars) and called `postupgrade` for the new canister code(with stable vars), so you need to put your migration code inside `postupgrade`, and, if needed, update the `preupgrade` code to be able upgrade canister in the future.

Also you can keep the old names for non-stable vars.

Reply:
Thanks this helped a lot. 
I updated the migration code and moved it into `postupgrade`.